### PR TITLE
Moved column types to smaller sub-components

### PR DESCRIPTION
**Pull request by damnko**

### CODE REVIEW

File: src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+@Component({
+  selector: 'cell-type-textarea',
+  template: `
+    <textarea [ngClass]="inputClass"
+              class="form-control"
+              [(ngModel)]="cell.newValue"
+              [name]="cell.getColumn().id"
+              [disabled]="!cell.getColumn().isEditable"
+              (click)="onClick($event)"
```
Comment (LexZhukov) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: - `(click)`, `(keydown.enter)` and other `@Output` parameters should be binding only with existing methods or using `(click)="onClick.emit($event)"` syntax.

Please, in future try to build the ng2-smart-table demo with AOT before pull-request.

**Comment by nnixaa** [comment]
Hey @damnko, a couple of technical comments below, hope they all make sense :)
@LexZhukov thanks for the review!

File: src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts (hunk 1 above)
Comment (damnko) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: Hi, thanks a lot for your review and excuse me for all the other code-formatting mistakes.

Regarding this comment, I think all those bindings were already there when I picked up the project, would you prefer if I create some methods in the components and bind to them like

_Template_
```
[name]="getId()"
```
_Component_
```
getId() {
  return this.cell.getColumn().id;
}
```
?

Regarding the "build with AOT" do you refer to running `npm run build:aot` before submitting the PR? If not, could you point me in the right direction? Thanks a lor

Comment (nnixaa) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: Hey @damnko, it's totally fine! With the first comment, the thing is that at the moment looks like you are calling non-exising onClick(..) method, when you need to call onClick property by doing like you did here https://github.com/akveo/ng2-smart-table/pull/144/files#diff-a56dc4441e739f28ca9a61ac3b73caddR14

For the second comment - the AOT mode has much more stricter rules, thus we just asking you to make sure the code is building even with AOT in the first place.

Comment (damnko) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: > With the first comment, the thing is that at the moment looks like you are calling non-exising onClick(..) method
That's true! It is non existing in fact... I will fix that, thanks for pointing that out, I didn't noticed it.

>  thus we just asking you to make sure the code is building even with AOT
Great to know, thanks for pointing that out. I will!

Comment (damnko) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: Actually, I am having problems with `npm run build:aot` but I am having them also in the `master` branch... 

It looks something like
```
Angular compilation done, starting webpack bundling.
Error: undefined, resolving symbol OpaqueToken in /*****/ng2-smart-table/node_modules/@angular/core/src/di/opaque_token.d.ts
...
```
Do you have any hint?

Comment (LexZhukov) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: @damnko can you remove `compiled`, `demo-dist` and `dll` folders and files like *.d.ts, *.js, *.js.map and *.metadata.json from project directory. Then try `npm run build:aot`.

Comment (damnko) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: Thanks it indeed helped to clean things up :) but I am still having that problem. The only `*.d.ts` file left is `custom-typings.d.ts`, all the rest that you suggested is gone... Have you got any other idea?

PS. Out of curiosity: is there a ready-to-use npm script (eg. `npm run clean`) that does all the above cleaning at once? Thanks!

Comment (LexZhukov) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: Try to remove node_modules folder and run `npm i`

Comment (damnko) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: Thanks a lot for your suggestions. I had to do all the above + update npm and node.
Btw everything compiles with no errors now so I am updating the PR.

Comment (nnixaa) [inline src/ng2-smart-table/components/cell/cell-types/cell-textarea.component.ts]: @damnko great to hear that!

**Pull request merged**